client/state/data-layer/wpcom/timezones/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import mapValues from 'lodash/mapValues';
+import map from 'lodash/map';
+import fromPairs from 'lodash/fromPairs';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import { TIMEZONES_REQUEST } from 'state/action-types';
+
+import {
+	timezonesRequestSuccess,
+	timezonesRequestFailure,
+	timezonesReceive,
+} from 'state/timezones/actions';
+
+/**
+ * Converts an array with the shape [ { value: 'foo', label: 'bar' } ]
+ * to a keyed object { foo: 'bar' }.
+ *
+ * @param  {Array} data - data array to convert
+ * @return {Object} a keyed object
+ */
+const valueLabelToObject = data => (
+	fromPairs( map( data, ( { label, value } ) => ( [ value, label ] ) ) )
+);
+
```

Comment:
Not sure if it seems clear to me whether or not this function is very generalizable or more specific. In the general sense it's `valueLabelToObject` but in the specific case inside of this module its feels more like it could be `timezoneRecordsToMap` or `apiTimezonesToObject`

Please don't change it just because I wrote this. I'm just making an observation and raising a question.